packages/test/test-end-to-end-tests/src/test/incrementalSummaryContext.spec.ts

```diff
@@ -0,0 +1,243 @@
+		const content = await readAndParse<ISnapshot>(storage, snapshotFileName);
+		for (const blob of content.blobs) {
+			const blobContent = await readAndParse<IBlob>(storage, blob);
+			this.blobMap.set(blob, blobContent);
+		}
+	}
+	protected processCore(
+		message: ISequencedDocumentMessage,
+		local: boolean,
+		localOpMetadata: unknown,
+	) {
+		if (message.type === MessageType.Operation) {
+			const op = message.contents as IOp;
+			switch (op.type) {
+				case "blobStorage": {
+					const blob: IBlob = {
+						value: op.value,
+						seqNumber: message.sequenceNumber,
+					};
+					const blobName = `${this.blobMap.size}`;
+					this.blobMap.set(blobName, blob);
+					break;
+				}
+				default:
+					throw new Error("Unknown operation");
+			}
+		}
+	}
+
+	public createOp(content: string) {
+		const op: IOp = {
+			type: "blobStorage",
+			value: content,
+		};
+		this.submitLocalMessage(op);
+	}
+
+	protected onDisconnect() {}
+	protected applyStashedOp(content: any): unknown {
+		throw new Error("Method not implemented.");
+	}
+}
+
+/**
+ * Validates w
+ */
+describeNoCompat(
+	"Incremental summary context fields are properly populated",
+	(getTestObjectProvider) => {
+		let provider: ITestObjectProvider;
+		const dataObjectFactory = new TestFluidObjectFactory([
+			["abc", TestSharedObject.getFactory()],
+		]);
+		const runtimeOptions: IContainerRuntimeOptions = {
+			summaryOptions: { summaryConfigOverrides: { state: "disabled" } },
+		};
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataObjectFactory,
+			[[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
+			undefined,
+			[innerRequestHandler],
+			runtimeOptions,
+		);
+
+		const createContainer = async (): Promise<IContainer> => {
+			return provider.createContainer(runtimeFactory);
+		};
+
+		async function createSummarizer(container: IContainer, summaryVersion?: string) {
+			const createSummarizerResult = await createSummarizerFromFactory(
+				provider,
+				container,
+				dataObjectFactory,
+				summaryVersion,
+				getContainerRuntimeApi(pkgVersion, pkgVersion)
+					.ContainerRuntimeFactoryWithDefaultDataStore,
+			);
+			return createSummarizerResult.summarizer;
+		}
+
+		beforeEach(async () => {
+			provider = getTestObjectProvider({ syncSummarizer: true });
+		});
+
+		it("works", async () => {
+			const container = await createContainer();
+			const datastore = await requestFluidObject<ITestFluidObject>(container, "default");
+			const dds = await datastore.getSharedObject<TestSharedObject>("abc");
+			dds.createOp("test data 1");
+			dds.createOp("test data 2");
+			dds.createOp("test data 3");
+
+			const summarizer = await createSummarizer(container);
+			await provider.ensureSynchronized();
+			await summarizeNow(summarizer);
+
+			dds.createOp("test data 4");
```

Comment:
Looks like this will go in a different blob and since its a new blob it would summarize but other blobs would be handles?
Again, comments :-).